ui/app/src/views/CreatePoolPage.vue

```diff
@@ -108,16 +109,33 @@ export default defineComponent({
         throw new Error("Token B field amount is not defined");
 
       transactionState.value = "signing";
-      let tx = await actions.clp.addLiquidity(
-        tokenBFieldAmount.value,
-        tokenAFieldAmount.value
-      );
 
-      console.log("POOL transaction hash: ", tx);
-      transactionHash.value = tx?.transactionHash ?? "";
-      transactionState.value = "confirmed";
+      try {
+        let tx = await actions.clp.addLiquidity(
+          tokenBFieldAmount.value,
+          tokenAFieldAmount.value
+        );
+
+        console.log("POOL transaction hash: ", tx);
+        transactionHash.value = tx?.transactionHash ?? "";
+        if (tx && tx.rawLog && tx.rawLog.includes('"type":"added_liquidity"')) {
+          transactionState.value = "confirmed";
+        } else {
+          transactionState.value = "failed";
+          transactionStateMsg.value = "Oops... Something went wrong. Please try again!";
+        }
 
-      clearAmounts();
+        clearAmounts();
+      } catch (e) {
+        // TODO: Implement better error checks and status updates. -> check swap also
+        if (e.toString().includes("Request rejected")) {
+          transactionState.value = "rejected";
+          transactionStateMsg.value = "Please confirm the transaction in your wallet.";
+        } else {
+          transactionState.value = "failed";
+          transactionStateMsg.value = "Oops... Something went wrong. Please try again!";
+        }
+      }
```

Comment:
Yeah so having this ceremony everywhere is what I was worried about from the perspective of my comment on @iverc 's PR 🤔 @novelja do you feel like simplifying this from the vue layer pushing some of this marshalling at least down to core and applying it to swap too?

Reply:
I think we could do better with:

```ts
let tx = await actions.clp.addLiquidity(
  tokenBFieldAmount.value,
  tokenAFieldAmount.value
);
transactionHash.value = tx.hash
transactionState.value = tx.state
```

Reply:
Or there is also a more event emitter driven approach where instead of awaiting 
```ts
const tx = actions.clp.addLiquidity(
  tokenBFieldAmount.value,
  tokenAFieldAmount.value
);

tx.onTxHash((txHash:string) => {
  transactionHash.value  = txHash;
});

tx.onStateChanged((state:Tx.State) => {
  transactionState.value  = txHash
}); 

tx.onError((err) => { 
  tellUserSomethingRealBadHasHappened(err);
});
```

Reply:
These can be combined in a "promievent" pattern where you can do both so I think if we support option1 now we can add optinon2 later and that wont involve changing code

Reply:
What I understand you might be saying would be we write to the store from within core and then have our view read from that instead of track the transaction in the handler. That makes sense to me.